### PR TITLE
TMEDIA-638 : Visual Regression

### PR DESCRIPTION
**Pull request by ecarlisle**
# Overview
This change runs Chromatic tests on pushes via a Github Action.

# Changes
1.  Added Chromatic as dev dependency.
2. Added `CHROMATIC_PROJECT_TOKEN` as a repository secret.
3. Created `/.github/workflows/chromatic.yml` Github actions to run Chromatic tests.

# Testing 
1. Rerun the latest run of the new Github Action: https://github.com/WPMedia/arc-themes-components/actions/workflows/chromatic.yml
2. Verify successful run in Chromatic: https://www.chromatic.com/builds?appId=61eeeb27215b5f003acdf18f

### CODE REVIEW

File: .github/workflows/chromatic.yml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+        uses: actions/cache@v2
+        with:
+          path: ~/.npm
+          key: npm-${{ hashFiles('**/package-lock.json') }}
+          restore-keys: |
+            restore-keys: npm-
+
+      - name: Install dependencies
+        if: steps.cache.outputs.cache-hit != 'true'
+        run: npm ic
```
Comment (JackHowa) [inline .github/workflows/chromatic.yml]: npm ci? 
```suggestion
        run: npm ci
```

File: .github/workflows/chromatic.yml [hunk 2]
```diff
@@ -0,0 +1,41 @@
+# .github/workflows/chromatic.yml
+
+name: 'Chromatic'
+
+on: push
```
Comment (JackHowa) [inline .github/workflows/chromatic.yml:5]: on push to all branches? In the past, we've done visual regression testing on pr 🤔

Comment (JackHowa) [reply .github/workflows/chromatic.yml:5]: not sure what the intended usecase is based on the ticket but this is good!

File: .github/workflows/chromatic.yml (hunk 1 above)
Comment (ecarlisle) [reply .github/workflows/chromatic.yml]: @JackHowa : Good catch! Got it, thanks!

File: .github/workflows/chromatic.yml [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+      - name: Cache dependencies
+        uses: actions/cache@v2
+        with:
+          path: ~/.npm
+          key: npm-${{ hashFiles('**/package-lock.json') }}
+          restore-keys: |
+            restore-keys: npm-
+
+      - name: Install dependencies
+        if: steps.cache.outputs.cache-hit != 'true'
```
Comment (JackHowa) [inline .github/workflows/chromatic.yml:32]: is this id steps cache ever set as true? If you see this example, you'll notice the id of the step is used for this logic https://github.com/WPMedia/engine-theme-sdk/blob/arc-themes-release-version-1.19/.github/workflows/pr-test.yml#L25

File: .github/workflows/chromatic.yml (hunk 2 above)
Comment (JackHowa) [reply .github/workflows/chromatic.yml:5]: cc @matthewroach

**Comment by ecarlisle** [comment]
@matthewroach : Also wondering about this failed run. Thanks!
https://github.com/WPMedia/arc-themes-components/runs/4929599787?check_suite_focus=true

File: .github/workflows/chromatic.yml (hunk 2 above)
Comment (matthewroach) [reply .github/workflows/chromatic.yml:5]: The action should be running on all pushes - it's the best way for chromatic to keep track of everything, and you need to run on head branches. Arc Theme Blocks is a good example - https://github.com/WPMedia/arc-themes-blocks/blob/arc-themes-release-version-1.19/.github/workflows/chromatic.yml

**Comment by matthewroach** [comment]
@ecarlisle From looking at the failure it looks to be an issue with what was installed as node modules, it looks like you are not caching the `node_modules` folder in the `with` part - the Arc Theme Blocks is a good example to follow - https://github.com/WPMedia/arc-themes-blocks/blob/arc-themes-release-version-1.19/.github/workflows/chromatic.yml#L39

**Review by JackHowa — DISMISSED** [review]
looks like the build is failing

File: .github/workflows/chromatic.yml (hunk 3 above)
Comment (ecarlisle) [reply .github/workflows/chromatic.yml:32]: Good call! I've that on line 23 now.

**Comment by JackHowa** [comment]
rec'd removing the cache logic if it was causing problems. if we follow the same syntax that's working in other repos, and not working here, then there may be something else at play 


```
      - name: Cache Node Modules
        id: cache
        uses: actions/cache@v2
        with:
          path: node_modules
          key: node-modules-${{ hashFiles('package-lock.json') }}

      - name: Clean install dependencies if lockfile changed
        if: steps.cache.outputs.cache-hit != 'true'
        run: npm ci
        env:
          NODE_AUTH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```

**Comment by matthewroach** [comment]
The issue was with the path used in the caching part of the action - we needed to only cache the `node_modules` folder - I updated it to verify and everything is running again now

**Review by matthewroach — DISMISSED** [review]

**Comment by ecarlisle** [comment]
> The issue was with the path used in the caching part of the action - we needed to only cache the `node_modules` folder - I updated it to verify and everything is running again now

@matthewroach : Looks like that was the issue. Thank you!
@JackHowa : Would you mind resolving the change request? Thanks!

**Review by matthewroach — APPROVED** [review]

**Review by JackHowa — APPROVED** [review]

**Pull request merged**